### PR TITLE
Handle reconnection

### PR DESCRIPTION
**Pull request by esdrasjnr**
**Describe what this PR introduces:** This patch add support to reconnection to the SDK

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce any breaking changes?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] Related issues are referenced in the PR (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing
- [ ] New/updated tests are included

**Testing environment:**

- Operating System/Platform: Ubuntu 18.04
- Node version: 12.15
- Yarn version: 1.22

If you have relevant logs, error output and etc, please attach them.

**Other information:**

### CODE REVIEW

File: src/Client.js [hunk 1]
```diff
@@ -64,15 +64,35 @@ class Client {
...
-    return this.amqp.stop();
+    const onDisconnected = async () => {
+      // clear consumers:
+      await Promise.all(
+        this.consumers.map(({ tag }) => this.amqp.unsubscribeConsumer(tag))
+      );
+      this.consumers = [];
+    };
+
+    return this.amqp.stop(onDisconnected);
```
Comment (netoax) [inline src/Client.js:95]: Why are you passing this callback if it doesn't react to any event?

File: src/Client.js [hunk 2]
```diff
@@ -64,15 +64,35 @@ class Client {
...
+    return this.amqp.start(onConnected);
   }
 
   async close() {
-    await Promise.all(
-      this.consumers.map(({ tag }) => this.amqp.unsubscribeConsumer(tag))
-    );
-    this.consumers = [];
-    return this.amqp.stop();
+    const onDisconnected = async () => {
```
Comment (netoax) [inline src/Client.js:87]: Is this change necessary?

File: src/network/AMQP.js [hunk 3]
```diff
@@ -11,20 +11,22 @@ export default class AMQP {
     };
   }
 
-  async start() {
+  async start(onConnection) {
```
Comment (netoax) [inline src/network/AMQP.js:14]:
```suggestion
  async start(onConnected) {
```

File: src/Client.js [hunk 4]
```diff
@@ -64,15 +64,35 @@ class Client {
...
+            topic,
+            `${topic}-${this.userKey}-${tag}`,
+            handler,
+            { consumerTag: tag }
+          )
+        )
+      );
+    };
+
+    return this.amqp.start(onConnected);
```
[the remark this answers is not in the document]
Comment (esdrasjnr) [reply src/Client.js]: Done!

File: src/Client.js (hunk 1 above)
Comment (esdrasjnr) [reply src/Client.js]: This was to maintain a standard in code. In any case I moved the code to another method!

File: src/Client.js (hunk 2 above)
Comment (esdrasjnr) [reply src/Client.js]: Change it!

File: src/network/AMQP.js (hunk 3 above)
Comment (esdrasjnr) [reply src/network/AMQP.js:14]: Done!
